### PR TITLE
Add Base32

### PR DESCRIPTION
**Pull request by SkeletalDemise**
Adds a Base32 decoder.

### CODE REVIEW

File: src/decoders/base32_decoder.rs [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+///! Call base64_decoder.crack to use. It returns option<String> and check with
+///! `result.is_some()` to see if it returned okay.
+///
+use super::interface::Crack;
+use super::interface::Decoder;
+
+use data_encoding::BASE32;
+use log::{debug, info, trace};
+
+/// The Base64 decoder, call:
```
Comment (bee-san) [inline src/decoders/base32_decoder.rs:16]: These docs are for base64 :)

File: src/decoders/base32_decoder.rs [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        Decoder {
+            name: "base32",
+            description: "Base32 is a group of binary-to-text encoding schemes that represent binary data (more specifically, a sequence of 8-bit bytes) in an ASCII string format by translating the data into a radix-64 representation.",
+            link: "https://en.wikipedia.org/wiki/Base32",
+            tags: vec!["base32", "decoder", "base"],
+            expected_runtime: 0.01,
+            expected_success: 1.0,
+            failure_runtime: 0.01,
+            normalised_entropy: vec![1.0, 10.0],
+            popularity: 1.0,
```
Comment (bee-san) [inline src/decoders/base32_decoder.rs:47]: I think base62 would probably be 0.7/0.8, 1.0 is for things like base64 / caesar (extremely common to see)

File: src/decoders/base32_decoder.rs [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+            assert_eq!(true, true);
+        }
+    }
+
+    #[test]
+    fn base32_handle_panic_if_emoji() {
+        let base32_decoder = Decoder::<Base32Decoder>::new();
+        let result = base32_decoder
+            .crack("😂", &get_athena_checker())
+            .unencrypted_text;
```
Comment (bee-san) [inline src/decoders/base32_decoder.rs:184]: Theoretically should we be able to decode an emoji with base32?

**Review by bee-san — CHANGES_REQUESTED** [review]

File: src/decoders/base32_decoder.rs (hunk 1 above)
Comment (SkeletalDemise) [reply src/decoders/base32_decoder.rs]: Fixed.

File: src/decoders/base32_decoder.rs (hunk 2 above)
Comment (SkeletalDemise) [reply src/decoders/base32_decoder.rs]: I changed it to `0.8`

File: src/decoders/base32_decoder.rs (hunk 3 above)
Comment (SkeletalDemise) [reply src/decoders/base32_decoder.rs:184]: No, it doesn't work.
https://gchq.github.io/CyberChef/#recipe=From_Base32('A-Z2-7%3D',false)&input=8J%2BYgg

**Review by bee-san — APPROVED** [review]

**Comment by bee-san** [comment]
🥳

**Pull request merged**